homeassistant/components/heatmiser/climate.py

```diff
@@ -5,72 +5,81 @@
 https://home-assistant.io/components/climate.heatmiser/
 """
 import logging
-
+import time
 import voluptuous as vol
 
 from homeassistant.components.climate import ClimateDevice, PLATFORM_SCHEMA
 from homeassistant.components.climate.const import (
-    SUPPORT_TARGET_TEMPERATURE)
+    STATE_AUTO, STATE_COOL, STATE_HEAT, STATE_ECO, SUPPORT_TARGET_TEMPERATURE,SUPPORT_OPERATION_MODE)
 from homeassistant.const import (
-    TEMP_CELSIUS, ATTR_TEMPERATURE, CONF_PORT, CONF_NAME, CONF_ID)
+    TEMP_CELSIUS, ATTR_TEMPERATURE, CONF_PORT, CONF_NAME, CONF_ID, PRECISION_WHOLE, STATE_OFF, STATE_ON, STATE_UNKNOWN)
+    
 import homeassistant.helpers.config_validation as cv
 
-REQUIREMENTS = ['heatmiserV3==0.9.1']
+REQUIREMENTS = ['heatmiserV3==1.1.8']
 
 _LOGGER = logging.getLogger(__name__)
 
 CONF_IPADDRESS = 'ipaddress'
+CONF_PORT = 'port'
 CONF_TSTATS = 'tstats'
 
-TSTATS_SCHEMA = vol.Schema({
-    vol.Required(CONF_ID): cv.string,
-    vol.Required(CONF_NAME): cv.string,
-})
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Required(CONF_IPADDRESS): cv.string,
-    vol.Required(CONF_PORT): cv.port,
-    vol.Required(CONF_TSTATS, default={}):
-        vol.Schema({cv.string: TSTATS_SCHEMA}),
-})
-
-
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the heatmiser thermostat."""
     from heatmiserV3 import heatmiser, connection
 
     ipaddress = config.get(CONF_IPADDRESS)
     port = str(config.get(CONF_PORT))
     tstats = config.get(CONF_TSTATS)
-
-    serport = connection.connection(ipaddress, port)
-    serport.open()
-
-    for tstat in tstats.values():
+    def_min_temp = 5
+    def_max_temp = 35
+    index = 0
+
+    uh1 = connection.HeatmiserUH1(ipaddress, port)
+
+    for tstat in tstats:
+        index += 1
+        room = tstat.get('room')
+        therm_id = tstat.get('id')
+        model = tstat.get('model')
+        sensor = tstat.get('sensor')
+        thermostat = heatmiser.HeatmiserThermostat(therm_id, model, uh1)
+        uh1_per = None
+        if index == 1:
+            uh1_per = uh1
+       
         add_entities([
             HeatmiserV3Thermostat(
-                heatmiser, tstat.get(CONF_ID), tstat.get(CONF_NAME), serport)
-            ])
-
-
+            thermostat, room, sensor, uh1_per, def_min_temp, def_max_temp)
+        ])
+        time.sleep(2.4)
```

Comment:
Using sleep() is usually a big no-no since it's blocking. What's the purpose of this sleep call?

Reply:
the heatmiser thermostats uses rs485 serial connection , and needs a 3 second delay between comms, the original component had this delay i've just tweaked it down as far as i can before the performance starts to suffer.